go/modelzooserver/modelzooserver.go

```diff
@@ -627,3 +667,51 @@ func (s *modelZooServer) DownloadModel(req *pb.ReleaseModelRequest, respStream p
 
 	return nil
 }
+
+func (s *modelZooServer) findLatestTag(name string) (string, error) {
+	findLatestVersionSQL := fmt.Sprintf(`SELECT id, version FROM %s WHERE name="%s" ORDER BY id DESC LIMIT 1;`, trainedModelTable, name)
+	rows, err := s.DB.Query(findLatestVersionSQL)
+	if err != nil {
+		return "", fmt.Errorf("execute SQL '%s' error: %v", findLatestVersionSQL, err)
+	}
+	var id int
+	var tag string
+	if !rows.Next() {
+		return "", fmt.Errorf("no model %s found", name)
+	}
+	err = rows.Scan(&id, &tag)
+	if err != nil {
+		return "", err
+	}
+	return tag, nil
+}
+
+func (s *modelZooServer) GetModelMeta(ctx context.Context, req *pb.ReleaseModelRequest) (*pb.GetModelMetaResponse, error) {
+	if os.Getenv("SQLFLOW_USE_EXPERIMENTAL_CODEGEN") != "true" {
+		return nil, fmt.Errorf("only support when SQLFLOW_USE_EXPERIMENTAL_CODEGEN=true")
+	}
+
+	if req.Tag == "" {
+		tag, err := s.findLatestTag(req.Name)
+		if err != nil {
+			return nil, err
+		}
+		req.Tag = tag
+	}
+
+	modelTableName := fmt.Sprintf("%s.%s_%s", publicModelDB, strings.ReplaceAll(req.Name, ".", "_"), strings.ReplaceAll(req.Tag, ".", "_"))
+	meta, err := experimental.GetModelMetadataFromDB(s.DB.URL(), modelTableName)
```

Comment:
Note that the current user may not have access to `modelTableName`, it may be trained by other users. We need to store the metadata in the model zoo table and get it directly. Mark this as a TODO for @typhoonzero .